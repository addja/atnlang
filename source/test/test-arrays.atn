def write_nl(a){
	print a; 
	print "%n";
}

def main() {
    x = 42;
	write_nl(x);
	x = true;
	write_nl(x);
	x[2] = 5;
	write_nl(x);
	x[1] = 2; 
	write_nl(x);
	x[9] = 100; 	
	write_nl(x);
	x[3] = true;	
	write_nl(x);
	x[1] = true; 	
	write_nl(x);
	x[2] = x[1];	
	write_nl(x);
	x[5] = false;	
	write_nl(x);
}

