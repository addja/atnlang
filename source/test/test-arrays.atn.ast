(PROGRAM (def write_nl (PARAMS a) (LIST_INSTR (print a) (print "%n"))) (def main PARAMS (LIST_INSTR (:= x 42) (FUNCALL write_nl (ARGLIST x)) (:= x true) (FUNCALL write_nl (ARGLIST x)) (:= ([ x 2) 5) (FUNCALL write_nl (ARGLIST x)) (:= ([ x 1) 2) (FUNCALL write_nl (ARGLIST x)) (:= ([ x 9) 100) (FUNCALL write_nl (ARGLIST x)) (:= ([ x 3) true) (FUNCALL write_nl (ARGLIST x)) (:= ([ x 1) true) (FUNCALL write_nl (ARGLIST x)) (:= ([ x 2) ([ x 1)) (FUNCALL write_nl (ARGLIST x)) (:= ([ x 5) false) (FUNCALL write_nl (ARGLIST x)))))