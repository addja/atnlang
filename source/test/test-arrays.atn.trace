main() <entry point>
|   write_nl(a=42) <line 8>
|   return <line 3>
|   write_nl(a=true) <line 10>
|   return <line 3>
|   write_nl(a=[0,0,5]) <line 12>
|   return <line 3>
|   write_nl(a=[0,2,5]) <line 14>
|   return <line 3>
|   write_nl(a=[0,2,5,0,0,0,0,0,0,100]) <line 16>
|   return <line 3>
|   write_nl(a=[false,false,false,true]) <line 18>
|   return <line 3>
|   write_nl(a=[false,true,false,true]) <line 20>
|   return <line 3>
|   write_nl(a=[false,true,true,true]) <line 22>
|   return <line 3>
|   write_nl(a=[false,true,true,true,false,false]) <line 24>
|   return <line 3>
return <line 3>
