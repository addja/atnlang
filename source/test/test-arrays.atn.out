42
true
[0,0,5]
[0,2,5]
[0,2,5,0,0,0,0,0,0,100]
[false,false,false,true]
[false,true,false,true]
[false,true,true,true]
[false,true,true,true,false,false]
